File: Java/src/jakopec/pomocno/Json.java
package jakopec.pomocno;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;
import jakopec.model.Mjesanje;
import jakopec.model.Partija;

import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class Json {

    public static void toJsonFile(String filePath, List<Partija> podaci){
        try {
            GsonBuilder gsonBilder = new GsonBuilder()
                    .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
                    .serializeNulls()
                    .setPrettyPrinting();
            Gson gson = gsonBilder.create();
            //System.out.println(podaci.size());
            System.out.println(gson.toJson(podaci));
            gson.toJson(podaci, new FileWriter(filePath)); // NE ZAPISUJE MI U FILE
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static List<Partija> fromJsonFile(String filePath){
        Type listType = new TypeToken<ArrayList<Partija>>(){}.getType();
        try {
            GsonBuilder gsonBilder = new GsonBuilder();
            gsonBilder.registerTypeAdapter(Partija.class, new PartijaAdapter());
            gsonBilder.registerTypeAdapter(Mjesanje.class, new MjesanjeAdapter());
            Gson gson = gsonBilder.create();
            return gson.fromJson(new FileReader(filePath), listType);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: Kotlin/src/main/kotlin/pomocno/Json.kt
package pomocno

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import model.Mjesanje
import model.Partija
import java.io.File
import java.io.FileNotFoundException
import java.io.FileReader
import java.io.IOException
import java.lang.reflect.Type

/**
 * @author Matija Skrletović
 **/

object Json {
    fun toJsonFile(filePath: String?, podaci: List<Partija>?) {
        try {
            val gsonBilder: GsonBuilder = GsonBuilder()
                .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
                .serializeNulls()
                .setPrettyPrinting()
            val gson: Gson = gsonBilder.create()
            val jsonPodaci: String = gson.toJson(podaci)
            File("podaci.json").writeText(jsonPodaci)

        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun fromJsonFile(filePath: String?): List<Partija>? {
        val listType: Type = object : TypeToken<ArrayList<Partija?>?>() {}.type
        try {
            val gsonBilder = GsonBuilder()
            gsonBilder.registerTypeAdapter(Partija::class.java, PartijaAdapter())
            gsonBilder.registerTypeAdapter(Mjesanje::class.java, MjesanjeAdapter())
            val gson: Gson = gsonBilder.create()
            return gson.fromJson<List<Partija>>(FileReader(filePath), listType)
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        }
        return null
    }
}